samples/word/99-preview-apis/compare-documents.yaml

```diff
@@ -0,0 +1,78 @@
+id: word-document-compare-documents
+name: Compare documents
+description: Compares two documents (the current one and a specified external one).
+author: YijunMS
+host: WORD
+api_set:
+    WordApi: '1.8'
+script:
+    content: |
+        $("#run").on("click", () => tryCatch(run));
+
+        async function run() {
+          // Compares the current document with a specified external document.
+          await Word.run(async (context) => {
+            // Absolute path of an online or local document.
+            const filePath = $("#filePath")
+              .val()
+              .toString();
+            // Options that configure the compare operation.
+            const options = { compareTarget: Word.CompareTarget.compareTargetCurrent };
+            context.document.compare(filePath, options);
+
+            await context.sync();
+
+            console.log("Differences shown in the current document.");
+          });
+        }
+
+        // Default helper for invoking an action and handling errors.
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            <p>This sample shows how to compare two documents (the current one and a specified external one).</p>
+        </section>
+
+        <section class="samples ms-font-m">
+            <h3>Try it out</h3>
+            <label>File (with absolute path): </label>
```

Comment:
Could this be a file picker instead?

Reply:
@AlexJerabek I'll look into that

Reply:
Looks like, for security reasons, can't get absolute path from the browser. The filename is real but nothing else - you get C:\fakepath\Document.docx